DBConnect.cc

```diff
@@ -54,4 +58,310 @@ void ConnectToMongoDB() {
     mongoc_cleanup();
 }
 
+bool SaveLayoutToDB(const std::string& name, const std::string& json_string) {
+    const char* uri_string = "mongodb://localhost:27017";
+    mongoc_uri_t* uri;
+    mongoc_client_t* client;
+    mongoc_database_t* database;
+    mongoc_collection_t* collection;
+    bson_t *command, reply, *insert, layout, *query;
+    bson_error_t error;
+    bool retval;
+    char* str;
+    char user[50];
+    bool result = true;
+
+    mongoc_init();
+
+    uri = mongoc_uri_new_with_error(uri_string, &error);
+    if (!uri) {
+        fprintf(stderr,
```

Comment:
Should we perhaps standardise on using `cout` and `cerr` everywhere rather than `fprintf`?

Reply:
We should use fmt::print for consistency, and cout if absolutely necessary.